onnxscript/rewriter/__init__.py

```diff
@@ -27,18 +25,17 @@ def rewrite(
     function_rewrite_rules: Sequence[type[FunctionRewriteRule]] = (),
     pattern_rewrite_rules: Sequence[PatternRewriteRule] = (),
 ) -> onnx.ModelProto:
+    print(f"len(value_info): {len(model.graph.value_info)}")
+    model_ir = ir.serde.deserialize_model(model)
```

Comment:
(Non-blocking comment: ok to create an issue to track this.) Maybe useful to support another scenario: in the context of the optimizer, we don't want to keep converting back and forth between IR and Proto in intermediate states ... we should do the conversion only once, in the beginning and end.

Supporting inputs which are both ModelProto and IR is easy and worth doing (just as a convenience for user really ... allowing IR input is important, allowing ModelProto input is convenience.)

For the output, however, we may need to make a choice. Either return IR always (reasonable first choice) or something more complicated (just as a convenience).

Reply:
https://github.com/microsoft/onnxscript/issues/1401